xbmc/epg/EpgInfoTag.cpp

```diff
@@ -263,9 +263,9 @@ float CEpgInfoTag::ProgressPercentage(void) const
   iDuration = endTime - startTime > 0 ? endTime - startTime : 3600;
 
   if (currentTime >= startTime && currentTime <= endTime)
-    fReturn = ((float) currentTime - startTime) / iDuration * 100;
```

Comment:
A 32-bit float can only represent integers till 24 bit long (without rounding), time_t of course exceeds this and this results in a miscalculation. So doing the time_t subtraction as int (and cast to float afterwards) will solve this.